x-pack/plugins/monitoring/server/kibana_monitoring/collectors/ops_buffer/ops_buffer.js

```diff
@@ -27,8 +27,14 @@ export function opsBuffer(server) {
     },
 
     flush() {
+      let cloud; // a property that will be left out of the result if the details are undefined
+      const cloudDetails = cloudDetector.getCloudDetails();
+      if (cloudDetails != null) {
+        cloud = { cloud: cloudDetails };
+      }
+
       return {
-        cloud: cloudDetector.getCloudDetails(),
+        ...cloud,
```

Comment:
For checking if payloads are empty, this field needed to be taken out when value is undefined